### DISCUSSION TITLE
Shut down on Mixxx exit

### DISCUSSION THREAD

**Discussion by MadTiXx** [fayaaz/mixxx-pi-gen#19] [Ideas]
Seems to me, like the raspberry pi still runs, after I exit Mixxx. 
Would make sense to shutdown raspberry pi after Closing Mixxx.

**Comment by fayaaz** [comment]
I hacked a way to do this which kind of works:

edit `.config/i3/config`

And change the line
```
exec devmon & /usr/bin/mixxx
```

to
```
exec devmon & /usr/bin/mixxx && sudo shutdown -h now
```
and when you quit Mixxx it should shut down.

**Reply by MadTiXx** [reply]
Thanks! :-) ( Imo the raspberry can damage sd-cards when it's not shut down before removing power source)

**Comment by aiaaaa** [comment]
also, you can use the following commands to have a graceful shutdown:

WIN + 2 (opens new window)
WIN + ENTER (opens terminal)
sudo halt

if you get good at this it looks really cool and takes less than 3 seconds, no mouse required

**Reply by fayaaz** [reply]
I thought in the latest version of the image if you hit `ctrl+Q` it would shut down the pi. Is this not the case?

**Reply by aiaaaa** [reply]
True, but by using a new window you have more options - you can substitute sudo halt for sudo reboot, if you way wanted to quickly reapply recently changed settings for instance. Your hack above, is elegant, and ensures data is not corrupted when using quit from the mixxx menu, but if you don't have access to the pi (if it is built into a case or in another room) the terminal method lets you choose the shutdown options. By the way really really cool project, I would love to help out however I can. I had added a (issue) question on HDMI audio I have been trying to troubleshoot the past few days.
